jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -5596,7 +5596,45 @@ static MethodHandle collectReturnValue(MethodHandle target, MethodHandle filter)
 		}
 		return buildTransformHandle(new CollectReturnHelper(target, filter), resultType);
 	}
-	
+
+	/*[IF JAVA_SPEC_VERSION >= 16]*/
+	/**
+	 * Helper class used by dropReturn​.
+	 */
+	private static final class DropReturnHelper implements ArgumentHelper {
+		private final MethodHandle target;
+
+		DropReturnHelper(MethodHandle target) {
+			this.target = target;
+		}
+
+		public Object helper(Object[] arguments) throws Throwable {
+			/* Invoke target. */
+			target.invokeWithArguments(arguments);
+			return null;
+		}
+	}
+
+	/**
+	 * Creates an adapter MethodHandle (MH) which drops the return value of the
+	 * target MH and has a void return type.
+	 * 
+	 * @param target represents the above target MH.
+	 * 
+	 * @return target if it already has a void return type; otherwise, an adapter MH.
+	 * @throws NullPointerException if target is null.
+	 */
+	public static MethodHandle dropReturn​(MethodHandle target) {
+		/* Implicit null check on target. */
+		MethodType targetType = target.type();
+		if (targetType.returnType() == void.class) {
+			return target;
+		}
+		targetType = targetType.changeReturnType(void.class);
+		return buildTransformHandle(new DropReturnHelper(target), targetType);
```

Comment:
I don't think we need this new helper class.  It should be possible to wrap the `target` with a `FilterReturnHandle` to remove the return value.

See the existing code in MethodHandle#asType

Reply:
Updated. I had initially thought of using `FilterReturnHandle` but I was stuck at creating the `filter` MH. Thanks for pointing me to the example.